src/txn/txn_rollback_to_stable.c

```diff
@@ -274,38 +327,49 @@ __rollback_row_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW
 
         /*
          * Stop processing when we find the newer version value of this key is stable according to
-         * the current version stop timestamp when it is not appending the selected update to the
-         * update chain. Also it confirms that history store doesn't contains any newer version than
-         * the current version for the key.
+         * the current version stop timestamp and transaction id when it is not appending the
+         * selected update to the update chain. Also it confirms that history store doesn't contains
+         * any newer version than the current version for the key.
          */
-        if (!replace && hs_stop_durable_ts <= rollback_timestamp) {
+        if (!replace &&
+          (!__rollback_check_if_txnid_non_committed(session, cbt->upd_value->tw.stop_txn)) &&
+          (hs_stop_durable_ts <= rollback_timestamp)) {
             __wt_verbose(session, WT_VERB_RECOVERY_RTS(session),
-              "history store update valid with stop timestamp: %s and stable timestamp: %s",
+              "history store update valid with stop timestamp: %s, stable timestamp: %s and txnid "
+              ": %" PRIu64,
               __wt_timestamp_to_string(hs_stop_durable_ts, ts_string[0]),
-              __wt_timestamp_to_string(rollback_timestamp, ts_string[1]));
+              __wt_timestamp_to_string(rollback_timestamp, ts_string[1]),
+              cbt->upd_value->tw.stop_txn);
             break;
         }
 
-        /* Stop processing when we find a stable update according to the given timestamp. */
-        if (hs_durable_ts <= rollback_timestamp) {
+        /*
+         * Stop processing when we find a stable update according to the given timestamp and
+         * transaction id.
+         */
+        if (!__rollback_check_if_txnid_non_committed(session, cbt->upd_value->tw.start_txn) &&
+          (hs_durable_ts <= rollback_timestamp)) {
             __wt_verbose(session, WT_VERB_RECOVERY_RTS(session),
               "history store update valid with start timestamp: %s, durable timestamp: %s, stop "
-              "timestamp: %s and stable timestamp: %s",
+              "timestamp: %s, stable timestamp: %s and txnid : %" PRIu64,
```

Comment:
No need of space after `txnid` better to be like `txnid:`

Reply:
Updated all the comments and removed unwanted white spaces.